api/index.js

```diff
@@ -31,6 +33,25 @@ function TaskAPIFactory (PouchDB) {
         taskStore.remove(taskDoc)
       }
 
+      api.progress = function (taskDoc, options) {
+        if (!taskDoc) throw new Error('Task Error requires taskdoc as first argument')
+
+        // shallow merge options with defaults
+        var progressState = _.assign({
+          name: 'progress',
+          at: new Date().toISOString()
+        }, options)
```

Comment:
I guess `_.assign` will work, too, but maybe [`_.defaults()`](https://lodash.com/docs/4.17.2#defaults) is a better fit for what we want, as it better expresses our intent?